### PR TITLE
Testing Ruby Code - Tic Tac Toe

### PR DESCRIPTION
**Pull request by Albertino2020**
In this project, we:

1. Added tests for all methods included in the Tic-tac-toe game project previously developed.
2. We added both positive and negative tests covering all possible scenarios
3. We checked for possible unexpected behaviors
3. We added a range of examples covering the correct and wrong usages of the methods.

Due to the fact that some tests in the original "player.rb" file should be done by receiving direct inputs from the players, we cloned the original bin and lib directories into bin_clone and lib_clone directories, respectively, so that we could disable the game's running functionalities and add inputs offline for the testing purposes.

Basically, the codes remain the same as the original ones, except that the inputs are not given by the players directly while playing, but are set specially for the testing purposes in the testing files.

All tests run have passed.

### CODE REVIEW

**Comment by Anna-Myzukina** [comment]
# Hi team ! :clap: 
### Good job   :eyes: 


# Status: [changes required]


* [x] All the public methods of the classes implemented in project OOP should be tested => from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)
screenshot 

* [x] A bit of common sense: all your specs should cover positive and negative scenarios, from  [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)
screenshot  
Example: there is a spec for correct usage of method [e.g. `it "creates object if all required attributes are passed"`]
and also there is spec for wrong usage of method  [e.g.` it "does not create object if required attributes are missing"`].

### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Comment by Albertino2020** [comment]
> # Hi team ! 
> ### Good job 
> # Status: [changes required]
> * [x]  All the public methods of the classes implemented in project OOP should be tested => from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)
>   screenshot
> * [ ]  A bit of common sense: all your specs should cover positive and negative scenarios, from  [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)
>   screenshot
>   Example: there is a spec for correct usage of method [e.g. `it "creates object if all required attributes are passed"`]
>   and also there is spec for wrong usage of method  [e.g.` it "does not create object if required attributes are missing"`].
> 
> ### Make changes and submit another code review request.
> _If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you._
> *Happy coding * computer

Hello Anna,
Would you please be more specific in which methods we didn't add tests for? Actually we have had tests for all public methods included in the Tic-tac-toe project? did you check it?

**Comment by Albertino2020** [comment]
> # Hi team ! 
> ### Good job 
> # Status: [changes required]
> * [ ]  All the public methods of the classes implemented in project OOP should be tested => from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)
>   screenshot
> * [x]  A bit of common sense: all your specs should cover positive and negative scenarios, from  [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)
>   screenshot
>   Example: there is a spec for correct usage of method [e.g. `it "creates object if all required attributes are passed"`]
>   and also there is spec for wrong usage of method  [e.g.` it "does not create object if required attributes are missing"`].
> 
> ### Make changes and submit another code review request.
> _If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you._
> *Happy coding * computer

We have added positive and negative tests too for all methods included in the Tic Tac Toe project. Didn't you see it?
One possible test that we may have step over is the last one you recommend that is to check if the methods don't create objects if not all correct parameters have been passed. But there are similar tests for the same effects. Did you notice them?

**Review by MarioDena — APPROVED** [review]
Congratulations on making the requested changes

## Project approved
